common/src/main/java/org/opensearch/ml/common/MLTask.java

```diff
@@ -150,6 +170,21 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(false);
         }
         out.writeBoolean(async);
+        if (streamOutputVersion.onOrAfter(MLTask.MINIMAL_SUPPORTED_VERSION_FOR_BATCH_TRANSFORM_JOB)) {
+            if (transformJob != null) {
+                out.writeBoolean(true);
+                try {
+                    AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+                        out.writeString(gson.toJson(transformJob));
```

Comment:
Is it ok to write a string to flat object field ?

Reply:
output.writeMap(transformJob, StreamOutput::writeString, StreamOutput::writeGenericValue)

Reply:
For Map<String, Object> type? I am trying to change transformJob's type to Map<String, String> but getting some exceptions while testing